### PR TITLE
Release/3.12.0

### PR DESCRIPTION
**Pull request by andrew-tran-infillion**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.DS_Store
+/.build
+/Packages
+xcuserdata/
+DerivedData/
+.swiftpm/configuration/registries.json
+.swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata
```
Comment (rblaa) [inline .gitignore:7]: why no the entire .swiftpm folder?

**Review by rblaa — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (andrew-tran-infillion) [reply .gitignore:7]: I think certain files in `.swiftpm` act like a `Package.lock` file for managing dependencies, so I wanted to keep those in.

Comment (rblaa) [reply .gitignore:7]: That makes sense.
However, I am not seeing them in the repo.
If you need them, they need to be added I think.

Comment (andrew-tran-infillion) [reply .gitignore:7]: Ah that's true, not really much in here. I think this was the boilerplate `.gitignore` set up for me by XCode though, so I'm a little hesitant to change it.

**Pull request closed without merging**